bin/alluxio

```diff
@@ -246,139 +160,172 @@ function copyDir {
   wait
 }
 
-PARAMETER=""
+function runJavaClass {
+  while getopts ":D:" opt; do
+    case ${opt} in
+    D)
+      ALLUXIO_SHELL_JAVA_OPTS+=" -D${OPTARG}"
+      ;;
+    \?)
+      echo "Invalid option: -${OPTARG}"
+      printUsage
+      exit 1
+      ;;
+    esac
+  done
+  shift $((${OPTIND} - 1))
+
+  ${JAVA} -cp ${CLASSPATH} ${ALLUXIO_USER_JAVA_OPTS} ${ALLUXIO_SHELL_JAVA_OPTS} ${CLASS} ${PARAMETER} $@
+}
+
+function main {
+
+  LAUNCHER=
+  # If debugging is enabled propagate that through to sub-shells
+  if [[ $- == *x* ]]; then
+    LAUNCHER="bash -x"
+  fi
+  BIN=$(cd "$( dirname "$0" )"; pwd)
+
+  if [[ $# == 0 ]]; then
+    printUsage
+    exit 1
+  fi
+
+  COMMAND=$1
+  shift
+
+  # bootstrap creates alluxio-env.sh
```

Comment:
add a comment that mentions that the needs to happens before source alluxio-config.sh